esp/xslt/esxdl2xsd.xslt

```diff
@@ -359,14 +359,15 @@
         </xsd:element>
     </xsl:template>
     <xsl:template match="EsdlService">
+        <xsl:variable name="useMethodName" select="@use_method_name='1'"/>
         <wsdl:message name="EspSoapFault">
             <wsdl:part name="parameters" element="tns:Exceptions"/>
         </wsdl:message>
         <xsl:for-each select="/esxdl/EsdlMethod">
             <wsdl:message>
                 <xsl:attribute name="name"><xsl:value-of select="@name"/>SoapIn</xsl:attribute>
                 <wsdl:part name="parameters">
-                    <xsl:attribute name="element">tns:<xsl:value-of select="@request_type"/></xsl:attribute>
+                    <xsl:attribute name="element">tns:<xsl:value-of select="@name"/><xsl:if test="not($useMethodName)">Request</xsl:if></xsl:attribute>
```

Comment:
This part looks fine as long as our requests are guaranteed to be labeled `<methodname>Request` - using `@name` or `@request_type` based on useMethodName seems slightly less problematic.

But more importantly, it seems the xsd:element definition referenced by this attribute needs to recognize useMethodName as well.

Reply:
@afishbeck it seems the EsdlRequest template (L:338) has no direct relationship to the method...
How do you feel about inferring the method name from the requestname?

`<xsl:attribute name="name"><xsl:value-of select="@name"/></xsl:attribute>`
to
`<xsl:attribute name="name><xsl:if test="$useMethodName"><xsl:value-of select="substring-before(@name, 'Request')"/></xsl:if><xsl:else><xsl:value-of select="@name"/></xsl:else></xsl:attribute>`

Reply:
@afishbeck bump

Reply:
@rpastrana updated.  I basically did what you suggested, just had to update the syntax a bit.

Reply:
@rpastrana Are you able to test?